alacritty/src/config/bindings.rs

```diff
@@ -319,6 +397,34 @@ pub fn default_key_bindings() -> Vec<KeyBinding> {
         F19; Action::Esc("\x1b[33~".into());
         F20; Action::Esc("\x1b[34~".into());
         NumpadEnter; Action::Esc("\n".into());
+        Escape, ModifiersState::CTRL;                              Action::ToggleKeyboardMode;
+        Escape,                        +TermMode::KEYBOARD_MOTION; Action::ToggleKeyboardMode;
+        I,                             +TermMode::KEYBOARD_MOTION; Action::ToggleKeyboardMode;
+        Q,                             +TermMode::KEYBOARD_MOTION; Action::ToggleKeyboardMode;
+        V,                             +TermMode::KEYBOARD_MOTION; Action::ToggleNormalSelection;
+        V,      ModifiersState::SHIFT, +TermMode::KEYBOARD_MOTION; Action::ToggleLineSelection;
+        V,      ModifiersState::CTRL,  +TermMode::KEYBOARD_MOTION; Action::ToggleBlockSelection;
+        V,      ModifiersState::ALT,   +TermMode::KEYBOARD_MOTION; Action::ToggleSemanticSelection;
+        Return,                        +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionClick;
+        K,                             +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionUp;
+        J,                             +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionDown;
+        H,                             +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionLeft;
+        L,                             +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionRight;
+        Key0,                          +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionStart;
+        Key4,   ModifiersState::SHIFT, +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionEnd;
+        G,                             +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionTop;
+        G,      ModifiersState::SHIFT, +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionBottom;
+        H,      ModifiersState::SHIFT, +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionHigh;
+        M,      ModifiersState::SHIFT, +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionMiddle;
+        L,      ModifiersState::SHIFT, +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionLow;
+        B,                             +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionPunctLeft;
+        W,                             +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionPunctRight;
+        E,                             +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionPunctRightEnd;
+        B,      ModifiersState::SHIFT, +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionWordLeft;
+        W,      ModifiersState::SHIFT, +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionWordRight;
+        E,      ModifiersState::SHIFT, +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionWordRightEnd;
+        Key5,   ModifiersState::SHIFT, +TermMode::KEYBOARD_MOTION; Action::KeyboardMotionBracket;
+        Y,                             +TermMode::KEYBOARD_MOTION; Action::Copy;
```

Comment:
I felt myself reaching for an equivalent of vim's ctrl-u and ctrl-d (scroll half-page) to help me move around my grid a bit faster.

Reply:
Bindings for entire buffer movement are already possible, at least Top/Bottom/PageUp/PageDown/LineUp/LineDown. I've decided to not add any bindings other than basic motion for now, since I think those are the most common. For everything else I think I'd suggest users to define their own bindings.

I think that ctrl+d for example is not a great binding and could easily lead to killed terminal sessions.

Reply:
How would I create a binding that moves the cursor up or down by half the page amount each time? I don't currently see a way to implement this, unless I'm being slow and missing something.

I haven't given conflicting keybindings any thought (yet), but in general I like following vi bindings for defaults though.

Reply:
@kchibisov just to be clear, `Ctrl + f` and `Ctrl + b` differ from the current default `Shift + H` and `Shift + L` because they will move up and down through history. Then the bindings `Ctrl + u` and `Ctrl + d` are the same as these bindings, just they move only half a page.

Ideally (I think) these are the same as the `PageUp` and a new `HalfPageUp` actions.